Framework/API/src/DetectorInfo.cpp

```diff
@@ -93,6 +93,20 @@ Kernel::Quat DetectorInfo::rotation(const size_t index) const {
   return getDetector(index).getRotation();
 }
 
+double DetectorInfo::eFixed(const size_t index) const {
+  const auto &detector = getDetector(index);
+  auto eFixedVector =
+      detector.getNumberParameter(Geometry::ParameterMap::eFixed());
+
+  double eFixed = 0.0;
+
+  if (!eFixedVector.empty()) {
+    eFixed = eFixedVector.front();
+  }
+
```

Comment:
I would be a bit unhappy about the return type for error reporting. Return values tend to be ignored...